extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -508,42 +508,57 @@ export const popupMessageListener = (request: Request) => {
     }
   };
 
-  const confirmPasswordHW = async () => {
```

Comment:
the `confirmPasswordHW` logic actually wasn't working (wasn't calling `login` in `confirmPasswordHW` oops). After looking over the two `confirmPassword` methods I think it's better to keep it as one to be more DRY and avoid future errors like this, so I removed the extra `confirmPasswordHW` function, and just refactored the original `confirmPassword` method to be able to handle hw or not